File: Tatsuki_Yabe/Q2/Swift/pseudocode_walkthrough.swift
class Solution {
    func validMountainArray(_ arr: [Int]) -> Bool {
        var i = 1
        while (i < arr.count && arr[i-1] < arr[i]) {
            i += 1
        }
        if (i == 1 || i == arr.count) {
            return false
        }
        while (i < arr.count && arr[i] < arr[i-1]) {
            i += 1
        }

        return i == arr.count
    }
}

File: Tatsuki_Yabe/Q2/Go/pseudocode_walkthrough.go
package main

func validMountainArray(arr []int) bool {
	i := 1
	for i < len(arr) && arr[i-1] < arr[i] {
		i++
	}
	for i == 1 || i == len(arr) {
		return false
	}
	for i < len(arr) && arr[i] < arr[i-1] {
		i++
	}

	return i == len(arr)
}
